web/packages/teleterm/src/ui/DocumentTerminal/Terminal/ctrl.ts

```diff
@@ -79,12 +84,85 @@ export default class TtyTerminal {
       },
     });
 
+    this.term.onSelectionChange(() => {
+      if (
+        this.configService.get('terminal.copyOnSelect').value &&
```

Comment:
`copyOnSelect` behaves a bit weird in combination with `rightClick: 'copyPaste'`.

When I select something and then press right click, the whole "word" gets selected and copied.

<details>
<summary>Comparison vidoes</summary>

`{ rightClick: 'copyPaste', copyOnSelect: true }`

https://github.com/user-attachments/assets/c472a794-1fb2-4e49-9d32-72f3f25a10b8

`{ rightClick: 'copyPaste', copyOnSelect: false }`


https://github.com/user-attachments/assets/cd3d168e-a06d-43ea-b9a1-df5eef496657
</details>

I think what's happening is that even with `copyOnSelect: false`, the right click seems to highlight the whole word but copies only the previously selected fragment. With `copyOnSelect: true`, the right click copies the selected text but it also changes the selection to the full word, which in turn copies the whole word.

idk if there's something we need to do about it. Maybe the config options should look a little bit different? Setting `{ rightClick: 'copyPaste', copyOnSelect: true }` at the same time is "legal" but idk if it makes much sense from the UX standpoint. Perhaps a smaller number of available choices would work well enough. I don't really want to dictate a solution here as I don't intuitively understand all requirements here.

https://github.com/user-attachments/assets/030f11f3-177a-4901-8126-cfde5f9cea55

Reply:
Good finding. 

To solve this problem, we could use the xterm.js's option `rightClickSelectsWord`.
For example, vscode sets this option to `false` when their `terminal.rightClick` is `selectWord`. When this option is active, right click selects a word and then opens the context menu. I don't think we need option too, instead we could set `rightClickSelectsWord` to `false` when  `terminal.rightClick` is `menu`. What do you think?

Reply:
> I don't think we need option too, instead we could set `rightClickSelectsWord` to `false` when `terminal.rightClick` is `menu`. What do you think?

| App | Shows context menu | Highlights word | Does copy/paste |
| --- | --- | --- | --- |
| cmd.exe | ❌ | ❌ | ✅ |
| Powershell | ❌ | ❌ | ✅ |
| Windows Terminal | ❌ | ❌ | ✅ |
| Terminal.app | ✅ | ✅ | ❌ |
| iTerm | ✅ | ✅ | ❌ |
| Terminal on Ubuntu | ✅ | ❌ | ❌ |
| Terminal on Fedora | ✅ | ❌ | ❌ |

Assuming we don't need to stick to differences between macOS and Linux, what do you say we do `rightClickSelectsWord: false` if `rightClick` is `paste` or `copyPaste` and `rightClickSelectsWord: true` if `rightClick` is `menu`?

We could consider `rightClickSelectsWord: false` when `rightClick: 'menu', copyOnSelect: true`, but I don't know what people would actually want here. We could leave it as is, I think `copyOnSelect` is something you care about mostly when `rightClick` is set to `pase` or `copyPaste`.

Reply:
>Assuming we don't need to stick to differences between macOS and Linux, what do you say we do rightClickSelectsWord: false if rightClick is paste or copyPaste and rightClickSelectsWord: true if rightClick is menu?

I agree, this makes sense. Actually, I wanted to propose the same in the comment above, but I totally mixed it up 🤦 
I wanted to say: 
> vscode sets this option to **true** when their terminal.rightClick is selectWord

>we could set rightClickSelectsWord to **true** when  terminal.rightClick is menu

And thanks for the table. It's really helpful, I appreciate the effort. 
I applied the change, and made the config to be read only when terminal starts. Otherwise, some parts of `ctrl.ts` would read the config dynamically (in event listeners), and some (`rightClickSelectsWord`) only in the constructor.


-----------
>We could consider rightClickSelectsWord: false when rightClick: 'menu', copyOnSelect: true, but I don't know what people would actually want here. We could leave it as is, I think copyOnSelect is something you care about mostly when rightClick is set to pase or copyPaste.

Agree 👍